crates/tabby-scheduler/src/lib.rs

```diff
@@ -53,25 +57,29 @@ pub async fn scheduler<T: RepositoryAccess + 'static>(now: bool, access: T) -> R
     Ok(())
 }
 
-fn job_index(repositories: &[RepositoryConfig]) {
+fn job_index(repositories: &[RepositoryConfig]) -> bool {
     println!("Indexing repositories...");
     let ret = index::index_repositories(repositories);
     if let Err(err) = ret {
         error!("Failed to index repositories, err: '{}'", err);
+        return false;
```

Comment:
instead of generate an `error!` here, use `return err.context("Failed to index repositories")`